src/Symfony/Component/Translation/Translator.php

```diff
@@ -372,6 +372,41 @@ private function doLoadCatalogue($locale)
                 $this->catalogues[$locale]->addCatalogue($this->loaders[$resource[0]]->load($resource[1], $locale, $resource[2]));
             }
         }
+
+        $this->resolveCatalogueMessages($this->catalogues[$locale]);
+    }
+
+    private function resolveCatalogueMessages($catalogue)
+    {
+        $expr = '/{([^}]++)}/';
+        foreach ($catalogue->all() as $domain => $messages) {
+            preg_match_all($expr, implode($messages), $matches);
+            $parameters = [];
+            foreach ($matches[1] as $key => $id) {
+                if (isset($messages[$id])) {
+                    $parameters[$matches[0][$key]] = $messages[$id];
+                }
+            }
+
+            if (!$parameters) {
+                continue;
+            }
+
+            while (preg_match($expr, implode($parameters))) {
+                foreach ($parameters as $id => $message) {
+                    $parameters[$id] = str_replace($matches[0], $parameters, $message);
```

Comment:
well, you have no guarentee that `$matches[0]` has no duplication, which would mean a different count than $parameters, which would break things :)